### PR TITLE
The image functions 

### PR DESCRIPTION
**Pull request by autoih**
typing of image functions (https://github.com/tensorflow/addons/issues/989)

### CODE REVIEW

File: tensorflow_addons/image/filters.py [hunk 1]
```diff
@@ -16,9 +16,13 @@
 import tensorflow as tf
 from tensorflow_addons.image import utils as img_utils
 from tensorflow_addons.utils import keras_utils
+from tensorflow_addons.utils.types import TensorLike
 
+from typing import Optional
 
-def _pad(image, filter_shape, mode="CONSTANT", constant_values=0):
+
+def _pad(image: TensorLike, filter_shape: TensorLike, mode: str = "CONSTANT",
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/filters.py]: filter_shape should be a list/tuple

File: tensorflow_addons/image/filters.py [hunk 2]
```diff
@@ -48,11 +52,11 @@ def _pad(image, filter_shape, mode="CONSTANT", constant_values=0):
 
 
 @tf.function
-def mean_filter2d(image,
-                  filter_shape=(3, 3),
-                  padding="REFLECT",
-                  constant_values=0,
-                  name=None):
+def mean_filter2d(image: TensorLike,
+                  filter_shape: TensorLike = [3, 3],
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/filters.py]: Same here

File: tensorflow_addons/image/filters.py [hunk 3]
```diff
@@ -116,11 +120,11 @@ def mean_filter2d(image,
 
 
 @tf.function
-def median_filter2d(image,
-                    filter_shape=(3, 3),
-                    padding="REFLECT",
-                    constant_values=0,
-                    name=None):
+def median_filter2d(image: TensorLike,
+                    filter_shape: TensorLike = [3, 3],
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/filters.py]: same here

File: tensorflow_addons/image/filters.py [hunk 4]
```diff
@@ -116,11 +120,11 @@ def mean_filter2d(image,
...
+                    padding: str = "REFLECT",
+                    constant_values: TensorLike = 0,
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/filters.py]:
```suggestion
                    constant_values: FloatTensorLike = 0,
```

File: tensorflow_addons/image/transform_ops.py [hunk 5]
```diff
@@ -32,8 +35,12 @@
 
 @tf.function
 def transform(
-    images, transforms, interpolation="NEAREST", output_shape=None, name=None
-):
+    images: TensorLike,
+    transforms: TensorLike,
+    interpolation: str = "NEAREST",
+    output_shape: Optional[str] = None,
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/transform_ops.py]:
```suggestion
    output_shape: Optional[list] = None,
```

File: tensorflow_addons/image/filters.py (hunk 2 above)
Comment (WindQAQ) [reply tensorflow_addons/image/filters.py]: it should be int or a list/tuple of int

**Comment by autoih** [comment]
Thanks @gabrieldemarmiesse and @WindQAQ.

File: tensorflow_addons/image/transform_ops.py [hunk 6]
```diff
@@ -255,7 +271,7 @@ def angles_to_projective_transforms(angles, image_height, image_width, name=None
 
 
 @tf.RegisterGradient("Addons>ImageProjectiveTransformV2")
-def _image_projective_transform_grad(op, grad):
+def _image_projective_transform_grad(op: TensorLike, grad: TensorLike) -> tf.Tensor:
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/transform_ops.py]: It's not needed to type the gradient functions. But if you really want to type it, op is a tensorflow operation (function), not a tensor.

Comment (autoih) [reply tensorflow_addons/image/transform_ops.py]: Thanks @gabrieldemarmiesse. I removed typing private function.

**Comment by gabrieldemarmiesse** [comment]
Thank you for the help! Thanks to you typing the public API takes way less time than I though it would!

**Comment by autoih** [comment]
Thank you a lot for the help of review, @gabrieldemarmiesse.

**Pull request merged**